editor/blocks/text/index.js

```diff
@@ -7,13 +7,14 @@ wp.blocks.registerBlock( 'core/text', {
 	category: 'common',
 
 	attributes: {
-		value: html()
+		value: query( 'p', html() )
```

Comment:
If you're only using the first paragraph, this can be simplified to:

```js
value: html( 'p' )
```

`query` will return an array of values scoped to each matching element for the selector, useful if we'd want to consider multiple paragraphs in this block.